core/src/main/java/com/kin/ecosystem/core/data/blockchain/BlockchainSourceImpl.java

```diff
@@ -81,7 +86,9 @@ private BlockchainSourceImpl(@NonNull EventLogger eventLogger, @NonNull final Ki
 		this.authRepository = authRepository;
 		this.kinClient = kinClient;
 		this.local = local;
-		this.activeAccountIndex = local.getAccountIndex();
+		Logger.log(new Log().withTag(TAG)
+			.put("BlockchainSourceImpl authRepository.getEcosystemUserID()", authRepository.getEcosystemUserID()));
+		this.currentUserId = authRepository.getEcosystemUserID();
```

Comment:
same I prefer one point of truth if you want caching to improve performance it should be handled by authRepository.getEcosystemUserID()

Reply:
it's not for cache this is the logic of blockchainsource.

Reply:
OK